### DISCUSSION TITLE
Midi Note to Frequency Bug?

### DISCUSSION THREAD

**Discussion by jonrichings** [pschatzmann/arduino-audio-tools#191] [Q&A] [answered]
In a few of the examples that require midi to frequency conversion MidiCommon::noteToFrequency(note); is called.  I found this didn't work because of a type problem.  It produces the same frequency for several notes.  
I'm not sure if this is an actual bug or just my incompetent use of the lib.

But I changed that line to 

```
float note = midiNote;
int frq  =  440 * pow(2, (note-69)/12);
```
And that works fine.

example:

```
void onNoteOn(uint8_t channel, uint8_t midiNote, uint8_t velocity) {
  float note = midiNote;
  //int frq = MidiCommon::noteToFrequency(note);
  int frq  =  440 * pow(2, (note-69)/12);
  Serial.print("onNoteOn: ");
  Serial.print(midiNote);
  Serial.print(": ");
  Serial.println(frq);
  sine.setFrequency(frq);
  adsr.keyOn();
}
```

**Comment by pschatzmann** [comment, marked as answer]
Thanks. I corrected the function
